nbgrader/server_extensions/formgrader/templates/gradebook_notebook_submissions.tpl

```diff
@@ -17,6 +17,11 @@ var notebook_id = "{{ notebook_id }}";
 </ol>
 {%- endblock -%}
 
+<!-- Button to toggle all student names -->
+{%- block messages -%}
+<button onclick="toggleAllNames()">Toggle student names</button>
```

Comment:
Can text be dynamic ?
Something like **Show all names** if some names are still hidden and **Hide all names** when all the names are already displayed.
This status could also define what should do the button (see previous comment)